lib/galaxy/webapps/tool_shed/model/__init__.py

```diff
@@ -246,10 +246,12 @@ def get_type_class( self, app ):
         return app.repository_types_registry.get_class_by_label( self.type )
 
     def get_tool_dependencies( self, changeset_revision ):
+        repo = hg.repository( ui.ui(), self.repo_path( app ) )
+        changeset_revision = suc.get_next_downloadable_changeset_revision( self, repo, changeset_revision )
```

Comment:
`get_next_downloadable_changeset_revision` is in `suc` in 16.04, but is in `metadata_util` in 16.07 and following releases. So you should retarget the PR to 16.04 or change this to `metadata_util`.